src/mesh_motion/FrameBase.C

```diff
@@ -67,9 +65,11 @@ void FrameBase::load(const YAML::Node& node)
 
 void FrameBase::populate_part_vec(const YAML::Node& node)
 {
-  // if nor parts specified and frame is inertial, return
-  if (!node["mesh_parts"] && isInertial_)
-    return;
+  // if no parts specified and frame is reference, return
+  if (!node["mesh_parts"]) {
+    throw std::runtime_error(
+      "FrameBase: No mesh parts found.");
+  }
```

Comment:
Does it make sense to allow user to transform the entire mesh if this is a one-time transformation action?

Reply:
@sayerhs This part of the code is common to both `mesh_transformation` and `mesh_motion`. My approach was to make sure that the user was more "aware" of what they are trying to move, but I can see how it might be a nuisance if there are a lot of parts involved and someone just wants to move all of them a certain way. I will change this, so let's not merge until I push the commit to allow this.

Reply:
@ashesh2512 Take a look at what @rcknaus had done for interior parts that allows user to specify all the parts with a single entry in the input file. May be something like that can be done here too that strikes the right balance between user convenience vs. error checking? 

https://github.com/Exawind/nalu-wind/blob/f539ccd12b4ce76e19382f34a53f9f2e3aa19a77/include/Realm.h#L691

Reply:
@sayerhs Thank you for the suggestions. I have added a commit to introduce this flexibility - 
https://github.com/Exawind/nalu-wind/pull/587/commits/2bf4c949c5c30a52b0daf02204377415fdde69f6